### PR TITLE
Adds Windows plugin documentation [SKIP CI]

### LINKED ISSUE #1664: Document Windows plugin
- [x] Installation:
A Windows plugin installation script has been added via PRs #1642, #1650 and #1654. We should also elaborate plugin installation instructions in the `README.md`, along with a few examples.
- [x] Known differences with vDVS on Linux:
a. Names
- [x] FAQs

/cc @shaominchen

### PR DESCRIPTION
**Pull request by venilnoronha**
This PR adds Windows plugin installation instructions to `README.md` and the user guide. This commit also adds a few general FAQs. Fixes #1664.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +187,10 @@ logging config format for content details.
 -  Volume is not shown when running ```docker volume ls``` and ```vmdkops_admin volume ls``` command after performing Storage vMotion of VM when volume is attached to VM. It is an expected behavior since Storage vMotion changes the location and name of attached vmdk files.
 We only see this when trying to do Storage vMotion with volume attached. So avoid Storage vMotion when volume is attached. [#1618](https://github.com/vmware/docker-volume-vsphere/issues/1618)
 
+## Known Differences Between Linux And Windows Plugin
+
+- Docker, by default, converts volume names to lower-case on Windows. Therefore, volume operations involving case-sensitive datastore names fail on Windows. [#1712](https://github.com/vmware/docker-volume-vsphere/issues/1712)
```
Comment (msterin) [inline README.md]: "may fail"

File: docs/user-guide/faq.md [hunk 2]
```diff
@@ -13,6 +13,12 @@ Please see README.md in the for the release by clicking on the tag for the relea
 ### How do I run the setup on my laptop?
 Follow the [guide on the wiki](https://github.com/vmware/docker-volume-vsphere/wiki/Using-laptop-for-running-the-entire-stack)
 
+### How do I bind a volume to a particular container host?
+This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere/documentation/tenancy.html).
+
+### Can I migrate data between Linux and Windows containers?
+Docker currently doesn't support migration of binaries between Windows and Linux, so container migration isn't possible. Also, volumes created via the Linux plugin are ext4 formatted by default, while the ones created via the Windows plugin are NTFS formatted, and they cannot be cross-mounted.
```
Comment (msterin) [inline docs/user-guide/faq.md]: binaries migration is not supported, but the question is not about binaries - it is about data migration. So while this `volumes created via the Linux plugin are ext4 formatted by default, while the ones created via the Windows plugin are NTFS formatted,` is accurate, this `they cannot be cross-mounted.` is not. *if* a Linux host has NTFS driver, e.g. http://www.tuxera.com/community/open-source-ntfs-3g/, they can mount the volume on Linux and use local volumes to provide the data into containers. Similarly, if they have ext4 support on Windows (e.g. http://www.ext2fsd.com/) they can mount the disk and then expose it to containers via local driver. What we want to say "while it is possible, our code does not support it automatically nor provides any explicit  help for these cases"

File: docs/user-guide/install.md [hunk 3]
```diff
@@ -91,3 +91,60 @@ vsphere
 ID                  NAME                DESCRIPTION         ENABLED
 ```
 
+## Installation on Windows Docker Hosts
```
Comment (msterin) [inline docs/user-guide/install.md]: we should not have the same text in 2 places. Readme.md can refer to this file. Please do not copy-n-paste chunks of docs

**Comment by venilnoronha** [comment]
@msterin addressed your comments in 729b54c.

File: README.md [hunk 4]
```diff
@@ -115,8 +120,9 @@ journalctl -fu docker.service # Journalctl/Systemd
 **VM (Docker-side) Plugin logs**
 
 * Log location: `/var/log/docker-volume-vsphere.log`
-* Config file location: `/etc/docker-volume-vsphere.conf`.
- * This JSON-formatted file controls logs retention, size for rotation
+* Config file location (Linux): `/etc/docker-volume-vsphere.conf`.
```
Comment (pshahzeb) [inline README.md]: Shouldn't we add log location on a Windows Host as well?

File: docs/user-guide/install.md [hunk 5]
```diff
@@ -91,3 +91,60 @@ vsphere
...
+PS C:\> .\install-vdvs.ps1 https://bintray.com/vmware/vDVS/download_file?file_path=docker-volume-vsphere_0.16_windows_tech_preview.zip
+Downloading from https://bintray.com/vmware/vDVS/download_file?file_path=docker-volume-vsphere_0.16_windows_tech_preview.zip...
+Extracting docker-volume-vsphere.zip into C:\Program Files\VMware\vmdkops...
+Deleting docker-volume-vsphere.zip...
+Installing Windows service vdvs from C:\Program Files\VMware\vmdkops\vdvs.exe...
+
+Status   Name               DisplayName
+------   ----               -----------
+Stopped  vdvs               vSphere Docker Volume Service
+Starting Windows service vdvs...
```
Comment (pshahzeb) [inline docs/user-guide/install.md]: Should we remove such verbose text from installation itself?
```Windows service vdvs installed successfully!```
Is quite enough

Comment (venilnoronha) [inline docs/user-guide/install.md]: It's planned, and we have created issue #1671 to track that.

File: README.md (hunk 4 above)
Comment (venilnoronha) [inline README.md]: Addressed in 657061c.

**Comment by venilnoronha** [comment]
The binary has been uploaded to bintray.

File: README.md [hunk 6]
```diff
@@ -39,8 +42,9 @@ Project page is located @ [https://vmware.github.io/docker-volume-vsphere/](http
 
 ## Supported Platform
 
-ESXi: 6.0 and above
-Docker: 1.12 and higher (Recommended 1.13/17.03 and above to use managed plugin)
+**ESXi:** 6.0 and above<br />
+**Docker (Linux):** 1.12 and higher (Recommended 1.13/17.03 and above to use managed plugin)<br />
+**Docker (Windows):** 17.03 and above (Windows containers mode only)
```
Comment (shuklanirdesh82) [inline README.md]: is there any dependency on docker 17.03? As per https://blog.docker.com/2016/09/build-your-first-docker-windows-server-container/ docker 1.12 should be fine too.

File: docs/user-guide/install.md [hunk 7]
```diff
@@ -91,3 +91,60 @@ vsphere
...
+* **To download the plugin installer**
+
+```
+PS C:\> Invoke-WebRequest -Uri https://raw.githubusercontent.com/vmware/docker-volume-vsphere/master/install-vdvs.ps1 -Out install-vdvs.ps1
+```
+
+* **To install the plugin**
+
+```
+PS C:\> .\install-vdvs.ps1 https://bintray.com/vmware/vDVS/download_file?file_path=docker-volume-vsphere_0.16_windows_developer-preview.zip
```
Comment (shuklanirdesh82) [inline docs/user-guide/install.md]: it is better to add a sample command as it keeps changing per release ... e.g. `PS C:\> .\install-vdvs.ps1 <windows_plugin bundle>`

File: README.md (hunk 6 above)
Comment (venilnoronha) [inline README.md]: I have only tested it with `17.03`. We could make it `1.12` if that works.

File: docs/user-guide/install.md (hunk 7 above)
Comment (venilnoronha) [inline docs/user-guide/install.md]: Addressed in f7bb3d8.

File: README.md (hunk 6 above)
Comment (shaominchen) [inline README.md]: Let's verify if 1.12 also works and update this part accordingly.

File: README.md [hunk 8]
```diff
@@ -114,9 +122,11 @@ journalctl -fu docker.service # Journalctl/Systemd
 
 **VM (Docker-side) Plugin logs**
 
-* Log location: `/var/log/docker-volume-vsphere.log`
-* Config file location: `/etc/docker-volume-vsphere.conf`.
- * This JSON-formatted file controls logs retention, size for rotation
+* Log location (Linux): `/var/log/docker-volume-vsphere.log`
+* Log location (Windows): `C:\Windows\System32\config\systemprofile\AppData\Local\docker-volume-vsphere\logs\docker-volume-vsphere.log`
```
Comment (shaominchen) [inline README.md]: Is this the right location for all supported windows versions? Ideally we should put logs somewhere under ProgramData, such as:

C:\ProgramData\VMware\vDVS\Logs\

File: docs/user-guide/docker-plugin-drivers.md [hunk 9]
```diff
@@ -19,7 +19,7 @@ The Docker photon volume driver supports provisioning and managing docker volume
 The Docker Volume plugin can support either or both types of volumes, as required, on a given Docker host.
 
 ## Configuring the Docker Volume Plugin
-The docker volume plugin loads runtime options and values from a json configuration file (default `/etc/docker-volume-vsphere.conf`) on the host. The user can override the default configuration by providing a different configuration file, via the `--config` option, specifying the full path of the file. Options that are currently recognized include the below set. Options passed on the command line override those in the configuration file.
+The docker volume plugin loads runtime options and values from a json configuration file (default `/etc/docker-volume-vsphere.conf` for Linux, `C:\ProgramData\docker-volume-vsphere\docker-volume-vsphere.conf` for Windows) on the host. The user can override the default configuration by providing a different configuration file, via the `--config` option, specifying the full path of the file. Options that are currently recognized include the below set. Options passed on the command line override those in the configuration file.
```
Comment (shaominchen) [inline docs/user-guide/docker-plugin-drivers.md]: Nits: json => JSON

File: docs/user-guide/faq.md [hunk 10]
```diff
@@ -13,6 +13,12 @@ Please see README.md in the for the release by clicking on the tag for the relea
 ### How do I run the setup on my laptop?
 Follow the [guide on the wiki](https://github.com/vmware/docker-volume-vsphere/wiki/Using-laptop-for-running-the-entire-stack)
 
+### How do I bind a volume to a particular container host?
```
Comment (shaominchen) [inline docs/user-guide/faq.md]: I think "attach" is more accepted than "bind"

File: README.md (hunk 8 above)
Comment (venilnoronha) [inline README.md]: This requires a code change, so should we do it another PR?

File: docs/user-guide/faq.md (hunk 10 above)
Comment (venilnoronha) [inline docs/user-guide/faq.md]: Here, by "bind" I mean that a volume is closely tied to a container host, and not in a literal "attach" sense. Do you suggest that we use "attach" still?

File: README.md (hunk 8 above)
Comment (shaominchen) [inline README.md]: Yes, we can file a separate PR to track the improvement of moving logs to C:\ProgramData\VMware\vDVS\Logs\. But can you please verify above location (C:\Windows\System32\config\...) is correct for both Windows 10 and Windows Server 2016, which we have announced official support in the document below?

File: docs/user-guide/faq.md (hunk 10 above)
Comment (shaominchen) [inline docs/user-guide/faq.md]: Then keep it as is.

File: README.md (hunk 8 above)
Comment (venilnoronha) [inline README.md]: This location is built upon the [`LOCALAPPDATA`](https://github.com/vmware/docker-volume-vsphere/blob/300327abfd03c94aafc226ca222904fb0562bc4e/client_plugin/utils/config/config_windows.go#L26) environment variable which returns `C:\Windows\System32\config\systemprofile\AppData\Local\` for Windows services, and hence that path.

Comment (shaominchen) [inline README.md]: Does LOCALAPPDATA environment variable have the same value on Windows 10 and Windows Server 2016?

File: docs/user-guide/docker-plugin-drivers.md (hunk 9 above)
Comment (venilnoronha) [inline docs/user-guide/docker-plugin-drivers.md]: Addressed in 1aee511.

File: README.md (hunk 6 above)
Comment (venilnoronha) [inline README.md]: I tried with version `1.12` on Windows, and it failed to locate the plugin itself. So, I'm keeping this as `1.13/17.03`.

File: README.md (hunk 8 above)
Comment (venilnoronha) [inline README.md]: Yes.

**Comment by venilnoronha** [comment]
Squashed and rebased with `master` in ab062de, to resolve merge conflict.

**Comment by venilnoronha** [comment]
@msterin yes, I've tested those.

**Comment by venilnoronha** [comment]
Another rebase with `master` in c0f8231, and renamed `docker-volume-vsphere` to `vsphere-storage-for-docker` at some places.

There are [2 places](https://github.com/vmware/vsphere-storage-for-docker/pull/1730/files#diff-d91c9bc00bf6167e14af3a6b3d1f887bR117) in `install-vdvs.ps1` where we still call out `docker-volume-vsphere`, and #1671 should fix that.

**Comment by venilnoronha** [comment]
Yet another rebase in 9598c5c to resolve conflicts.
